src/lavinmq/vhost.cr

```diff
@@ -369,18 +369,68 @@ module LavinMQ
       upstreams.stop_all
     end
 
+    private def close_connections(reason, close_done : Channel(Nil))
+      ch = Channel(Client).new
```

Comment:
use a buffered `Channel` instead? And add fibers if `wg.wait` times out? would it result in less code? 

```suggestion
      ch = Channel(Client).new(@connections.size)
```

Reply:
Hm, not that easy. No timeout in `WaitGroup#wait`.

The waitgroup is make sure we're not returning from `VHost#close` before all `Client#close` is done.
The timeout to send is to know that we're not stuck with closing.

I think a change would require something like

```crystal
@connections.times do 
  loop do
    select
    when connection_closed.recieve?
      break
    when timout 100.milliseconds
      spawn_close_fiber.call
    end
  end
end
```

together with a `connection_closed.send` after `Client#close`.

Or am I missing something? 🤔

Reply:
ok